pynestml/codegeneration/nest_printer.py

```diff
@@ -471,3 +471,28 @@ def print_buffer_declaration_header(cls, ast_buffer):
         assert isinstance(ast_buffer, VariableSymbol), \
             '(PyNestML.CodeGeneration.Printer) No or wrong type of ast_buffer symbol provided (%s)!' % type(ast_buffer)
         return '//!< Buffer incoming ' + ast_buffer.get_type_symbol().get_symbol_name() + 's through delay, as sum'
+
+    def print_vector_declaration(self, variable: VariableSymbol) -> str:
+        """
+        Prints the vector declaration
+        :param variable: Vector variable
+        :return: the corresponding vector declaration statement
+        """
+        assert isinstance(variable, VariableSymbol), \
+            '(PyNestML.CodeGeneration.Printer) No or wrong type of variable symbol provided (%s)!' % type(variable)
+        vector_parameter = variable.get_vector_parameter()
+        vector_variable = ASTVariable(vector_parameter, scope=variable.get_corresponding_scope())
```

Comment:
Do we need this temporary variable here?

Reply:
This temporary variable is for the case where the vector parameter (here the size of the vector) is in itself a variable. This is needed to resolve the vector parameter variable to a symbol.